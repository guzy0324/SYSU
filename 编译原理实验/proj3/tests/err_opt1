/* Optional test case 1: array dimension misuse */
program erropt1;
class myTest
{
	declarations
		int[] x=int[10], y;
		int[][] z=int[3][4];
	enddeclarations

	method void main()
	{
	x[1][2]:=3;
	y[3]:=4;
	z[5]:=1;
	}

}
