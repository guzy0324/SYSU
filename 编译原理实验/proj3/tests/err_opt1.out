Semantic Error--line: 17, symbol x: has incorrect number of dimensions.
Semantic Error--line: 17, symbol z: has incorrect number of dimensions.
********************************Symbol Table************************************

          Name Nest-Level  Tree-Node Predefined        Kind      Type      Value     Offset  Dimension     Argnum

  1     myTest          0                            class                                                       
  2          y          1                            array  type-tree                                1           
  3          x          1                            array  type-tree                                1           
  4          z          1                            array  type-tree                                2           
  5       main          1  proc-tree             procedure                                                      0
************* SYNTAX TREE PRINTOUT ***********

  +-[DUMMYnode]
R-[ProgramOp]
  |   +-[STNode,1,"myTest"]
  | +-[ClassDefOp]
  | | |         +-[NUMNode,1]
  | | |       +-[AssignOp]
  | | |       | |     +-[DUMMYnode]
  | | |       | |   +-[SelectOp]
  | | |       | |   | | +-[DUMMYnode]
  | | |       | |   | +-[IndexOp]
  | | |       | |   |   +-[NUMNode,5]
  | | |       | | +-[VarOp]
  | | |       | | | +-[STNode,4,"z"]
  | | |       | +-[AssignOp]
  | | |       |   +-[DUMMYnode]
  | | |     +-[StmtOp]
  | | |     | |   +-[NUMNode,4]
  | | |     | | +-[AssignOp]
  | | |     | | | |     +-[DUMMYnode]
  | | |     | | | |   +-[SelectOp]
  | | |     | | | |   | | +-[DUMMYnode]
  | | |     | | | |   | +-[IndexOp]
  | | |     | | | |   |   +-[NUMNode,3]
  | | |     | | | | +-[VarOp]
  | | |     | | | | | +-[STNode,2,"y"]
  | | |     | | | +-[AssignOp]
  | | |     | | |   +-[DUMMYnode]
  | | |     | +-[StmtOp]
  | | |     |   |   +-[NUMNode,3]
  | | |     |   | +-[AssignOp]
  | | |     |   | | |       +-[DUMMYnode]
  | | |     |   | | |     +-[SelectOp]
  | | |     |   | | |     | | +-[DUMMYnode]
  | | |     |   | | |     | +-[IndexOp]
  | | |     |   | | |     |   +-[NUMNode,2]
  | | |     |   | | |   +-[SelectOp]
  | | |     |   | | |   | | +-[DUMMYnode]
  | | |     |   | | |   | +-[IndexOp]
  | | |     |   | | |   |   +-[NUMNode,1]
  | | |     |   | | | +-[VarOp]
  | | |     |   | | | | +-[STNode,3,"x"]
  | | |     |   | | +-[AssignOp]
  | | |     |   | |   +-[DUMMYnode]
  | | |     |   +-[StmtOp]
  | | |     |     +-[DUMMYnode]
  | | |   +-[BodyOp]
  | | |   | +-[DUMMYnode]
  | | | +-[MethodOp]
  | | | | |   +-[DUMMYnode]
  | | | | | +-[SpecOp]
  | | | | | | +-[DUMMYnode]
  | | | | +-[HeadOp]
  | | | |   +-[STNode,5,"main"]
  | | +-[BodyOp]
  | |   |         +-[INTEGERTNode]
  | |   |       +-[ArrayTypeOp]
  | |   |       | | +-[NUMNode,4]
  | |   |       | +-[BoundOp]
  | |   |       |   | +-[NUMNode,3]
  | |   |       |   +-[BoundOp]
  | |   |       |     +-[DUMMYnode]
  | |   |     +-[CommaOp]
  | |   |     | |     +-[DUMMYnode]
  | |   |     | |   +-[IndexOp]
  | |   |     | |   | +-[DUMMYnode]
  | |   |     | | +-[IndexOp]
  | |   |     | | | +-[DUMMYnode]
  | |   |     | +-[TypeIdOp]
  | |   |     |   +-[INTEGERTNode]
  | |   |   +-[CommaOp]
  | |   |   | +-[STNode,4,"z"]
  | |   | +-[DeclOp]
  | |   | | +-[DUMMYnode]
  | |   +-[BodyOp]
  | |     |       +-[DUMMYnode]
  | |     |     +-[CommaOp]
  | |     |     | |   +-[DUMMYnode]
  | |     |     | | +-[IndexOp]
  | |     |     | | | +-[DUMMYnode]
  | |     |     | +-[TypeIdOp]
  | |     |     |   +-[INTEGERTNode]
  | |     |   +-[CommaOp]
  | |     |   | +-[STNode,2,"y"]
  | |     | +-[DeclOp]
  | |     | | |       +-[INTEGERTNode]
  | |     | | |     +-[ArrayTypeOp]
  | |     | | |     | | +-[NUMNode,10]
  | |     | | |     | +-[BoundOp]
  | |     | | |     |   +-[DUMMYnode]
  | |     | | |   +-[CommaOp]
  | |     | | |   | |   +-[DUMMYnode]
  | |     | | |   | | +-[IndexOp]
  | |     | | |   | | | +-[DUMMYnode]
  | |     | | |   | +-[TypeIdOp]
  | |     | | |   |   +-[INTEGERTNode]
  | |     | | | +-[CommaOp]
  | |     | | | | +-[STNode,3,"x"]
  | |     | | +-[DeclOp]
  | |     | |   +-[DUMMYnode]
  | |     +-[BodyOp]
  | |       +-[DUMMYnode]
  +-[ClassOp]
    +-[DUMMYnode]
